Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -736,7 +735,7 @@ void WebsiteDataStore::fetchDataAndApply(OptionSet<WebsiteDataType> dataTypes, O
     }
 
     if (dataTypes.contains(WebsiteDataType::DeviceIdHashSalt)) {
-        m_deviceIdHashSaltStorage->getDeviceIdHashSaltOrigins([callbackAggregator](auto&& origins) {
+        ensureDeviceIdHashSaltStorage().getDeviceIdHashSaltOrigins([callbackAggregator](auto&& origins) {
```

Comment:
Ideally those would be calling `ensureProtectedDeviceIdHashSaltStorage()` too since you're calling a non-trivial member function on it. That said, the previous code wasn't following our smart pointer guidelines yet. So at least your PR is not regressing things in this aspect.